lib/hanami/components/components.rb

```diff
@@ -177,9 +177,11 @@ module Components # rubocop:disable Metrics/ModuleLength
       end
 
       resolve do |configuration|
-        Hanami::Mailer.configuration = Hanami::Mailer::Configuration.new if Hanami.code_reloading?
-        Hanami::Mailer.configure(&configuration.mailer)
-        Hanami::Mailer.configuration
+        if configuration.mailer
```

Comment:
@davydovanton I think should be like this line https://github.com/hanami/hanami/blob/9863c0949930f93155ed9362e2ea27bf732498c6/lib/hanami/components/components.rb#L105